package/libgluonutil/src/libgluonutil.c

```diff
@@ -109,6 +109,31 @@ struct json_object * gluonutil_wrap_and_free_string(char *str) {
 	return ret;
 }
 
+
+bool gluonutil_get_mesh_prefix(struct in6_addr *result) {
+	struct json_object *site = gluonutil_load_site_config();
+	if (!site)
+		return false;
+
+	struct json_object *node_prefix = NULL;
+	if (!json_object_object_get_ex(site, "node_prefix6", &node_prefix))
+		return false;
+
+	char *str_prefix = (char*)json_object_get_string(node_prefix);
+	if (!str_prefix)
+		return false;
+
+	char *prefix_addr = strndup(str_prefix, strchrnul(str_prefix, '/')-str_prefix);
+
+	inet_pton(AF_INET6, prefix_addr, &(result->s6_addr));
```

Comment:
You should also check the return value of *inet_pton* and only return true on success.